### PR TITLE
Augment default metrics

### PR DESCRIPTION
**Pull request by etnt93**
We'd like to see the ability to add custom labels to default metrics. For example, adding 
`labels['platform'] = env['HTTP_X_PLATFORM']` to `http_requests_total`. 

Also added 'queue' as a label for default sidekiq metrics in order to view metrics on sidekiq jobs at the 'queue' level.

### CODE REVIEW

**Comment by SamSaffron** [comment]
This is conflicting now, can you have a look?

**Comment by etnt93** [comment]
Resolved merge conflicts!

File: lib/prometheus_exporter/middleware.rb [hunk 1]
```diff
@@ -44,14 +44,25 @@ def call(env)
...
+    if labels.present? && labels.length > 0
+      obj = obj.merge(custom_labels: labels)
+    end
+
+    @client.send_json(obj)
+  end
+
+  # allows subclasses to add custom labels based on env
+  def custom_labels(env)
+    {}
```
Comment (SamSaffron) [inline lib/prometheus_exporter/middleware.rb:65]: let's return nil here, then we can avoid all this merging in the client when labels are not configured.

Also present? is going to be an active support thing I think

Comment (etnt93) [reply lib/prometheus_exporter/middleware.rb]: Returns `nil` and removed `present?` in the latest commit.

File: lib/prometheus_exporter/client.rb [hunk 2]
```diff
@@ -107,7 +107,7 @@ def find_registered_metric(name, type: nil, help: nil)
     end
 
     def send_json(obj)
-      payload = @custom_labels.nil? ? obj : obj.merge(custom_labels: @custom_labels)
+      payload = @custom_labels.nil? ? obj : obj.merge(custom_labels: @custom_labels.merge(obj[:custom_labels] || {}))
```
Comment (SamSaffron) [inline lib/prometheus_exporter/client.rb:110]: minor but this is perf sensitive. No need to merge here if obj[:custom_labels] is nil. 

```
payload = 
  if @custom_labels
     if obj[:custom_labels]
         ...
     else
        ...
     end 
  else
    obj
  end
  
```

File: lib/prometheus_exporter/middleware.rb [hunk 3]
```diff
@@ -44,14 +44,25 @@ def call(env)
...
+    }
+    labels = custom_labels(env)
+    if labels
+      obj = obj.merge(custom_labels: labels)
+    end
+
+    @client.send_json(obj)
+  end
+
+  # allows subclasses to add custom labels based on env
```
Comment (SamSaffron) [inline lib/prometheus_exporter/middleware.rb:63]: very important ... we are making an api change here so we must add something to the README about how to consume this.

**Comment by SamSaffron** [comment]
cool let's try this out... also our CI is playing up, got to get that sorted.

**Pull request merged**